mava/wrappers/jaxmarl.py

```diff
@@ -324,6 +325,26 @@ def __init__(
         add_agent_ids_to_state: bool = False,
     ):
         super().__init__(env, has_global_state, timelimit, add_agent_ids_to_state)
+        self.log_win_rate = self._env.name in ["HeuristicEnemySMAX", "LearnedPolicyEnemySMAX"]
+
+    def reset(
+        self, key: PRNGKey
+    ) -> Tuple[JaxMarlState, TimeStep[Union[Observation, ObservationGlobalState]]]:
+        state, ts = super().reset(key)
+        extras = {"won_episode": False} if self.log_win_rate else {}
+        ts = ts.replace(extras=extras)
+        return state, ts
+
+    def step(
+        self, state: JaxMarlState, action: Array
+    ) -> Tuple[JaxMarlState, TimeStep[Union[Observation, ObservationGlobalState]]]:
+        state, ts = super().step(state, action)
+        current_winner = jax.lax.select(
+            ts.step_type == StepType.LAST, jnp.all(ts.reward >= 1.0), False
+        )
```

Comment:
Avoids the select, I'm pretty sure this is the same thing, but please double check my logic:

```suggestion
        current_winner = (ts.step_type == StepType.LAST) & jnp.all(ts.reward >= 1.0)
```